R/Instance.R

```diff
@@ -1,166 +1,108 @@
-create_instance_class <- function(instance_settings) {
-  super <- NULL # satisfy R CMD check and lintr
+create_instance <- function(instance_settings) {
+  super <- NULL # satisfy linter
 
-  name <- paste0(instance_settings$owner, "/", instance_settings$name)
-  schema <- api_get_schema(instance_settings)
+  api <- API$new(instance_settings = instance_settings)
 
-  # use lapply instead of map to retain names
-  classes <- map(
-    names(schema),
-    function(module_name) {
-      model_names <- names(schema[[module_name]])
-      map(
-        model_names,
-        function(model_name) {
-          list(
-            module_name = module_name,
-            model_name = model_name,
-            class_name = schema[[module_name]][[model_name]]$class_name
-          )
-        }
-      )
+  # fetch schema from the API
+  schema <- api$get_schema()
+
+  # create active fields for the exposed instance
+  active <- list()
+
+  # add core models to active fields
+  for (model_name in names(schema$core)) {
+    model <- schema$core[[model_name]]
+
+    if (model$is_link_table) {
+      next
     }
-  ) |>
-    list_flatten() |>
-    transpose()
 
-  active <- pmap(
-    classes,
-    function(module_name, model_name, class_name) {
-      fun <- NULL # satisfy R CMD check and lintr
-      fun_src <- paste0(
-        "fun <- function() {\n",
-        "  private$classes[['", module_name, "']][['", model_name, "']]\n",
-        "}\n"
-      )
-      eval(parse(text = fun_src))
-      fun
+    fun <- NULL
+    fun_src <- paste0(
+      "fun <- function() {",
+      "  private$.module_classes$core$get_model('", model_name, "')",
+      "}"
+    )
+    eval(parse(text = fun_src))
+
+    active[[model$class_name]] <- fun
+  }
+
+  # add non-core modules to active fields
+  for (module_name in names(schema)) {
+    if (module_name == "core") {
+      next
     }
-  ) |>
-    set_names(classes$class_name)
 
-  instance_class <- R6::R6Class(
-    name,
+    fun <- NULL
+    fun_src <- paste0(
+      "fun <- function() {",
+      "  self$get_module('", module_name, "')",
+      "}"
+    )
+
+    eval(parse(text = fun_src))
+
+    active[[module_name]] <- fun
+  }
+
+  # create the instance class
+  RichInstance <- R6::R6Class( # nolint object_name_linter
```

Comment:
Right. I'm sortof using Scala terminology here, where a Rich Wrapper is used to denote a class which serves to extend an already existing class with extra fields. The terminology 'Instance' and 'BaseInstance' instead of 'RichInstance' and 'Instance' works for me too :) WDYT?